lib/active_merchant/billing/gateways/cyber_source_rest.rb

```diff
@@ -0,0 +1,221 @@
+require 'active_merchant/billing/gateways/cyber_source/cyber_source_common'
+
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class CyberSourceRestGateway < Gateway
+      include ActiveMerchant::Billing::CyberSourceCommon
+
+      self.test_url = 'https://apitest.cybersource.com'
+      self.live_url = 'https://api.cybersource.com'
+
+      self.supported_countries = ActiveMerchant::Billing::CyberSourceGateway.supported_countries
+      self.default_currency = 'USD'
+      self.supported_cardtypes = %i[visa master american_express discover diners_club jcb maestro elo union_pay cartes_bancaires mada]
+
+      self.homepage_url = 'http://www.cybersource.com'
+      self.display_name = 'CyberSourceRest'
+
+      CREDIT_CARD_CODES = {
+        american_express: '003',
+        cartes_bancaires: '036',
+        dankort: '034',
+        diners_club: '005',
+        discover: '004',
+        elo: '054',
+        jcb: '007',
+        maestro: '042',
+        master: '002',
+        unionpay: '062',
+        visa: '001'
+      }
+
+      def initialize(options = {})
+        requires!(options, :merchant_id, :public_key, :private_key)
+        super
+      end
+
+      def purchase(money, payment, options = {})
+        authorize(money, payment, options) do |post|
+          post[:processingInformation] = { capture: true }
+        end
+      end
+
+      def authorize(money, payment, options = {})
+        post = build_auth_request(money, payment, options)
+
+        yield post if block_given?
+
+        commit('/pts/v2/payments/', post)
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(/(\\?"number\\?":\\?")\d+/, '\1[FILTERED]').
+          gsub(/(\\?"securityCode\\?":\\?")\d+/, '\1[FILTERED]').
+          gsub(/(signature=")[^"]*/, '\1[FILTERED]').
+          gsub(/(keyid=")[^"]*/, '\1[FILTERED]').
+          gsub(/(Digest: SHA-256=)[\w\/\+=]*/, '\1[FILTERED]')
+      end
+
+      private
+
+      def build_auth_request(amount, payment, options)
+        { clientReferenceInformation: {}, paymentInformation: {}, orderInformation: {} }.tap do |post|
+          add_customer_id(post, options)
+          add_code(post, options)
+          add_credit_card(post, payment)
+          add_amount(post, amount)
+          add_address(post, payment, options[:billing_address], options, :billTo)
+          add_address(post, payment, options[:shipping_address], options, :shipTo)
+        end.compact
+      end
+
+      def add_code(post, options)
+        return unless options[:order_id].present?
+
+        post[:clientReferenceInformation][:code] = options[:order_id]
```

Comment:
I think this would be safe without the guard clause?
```suggestion
        post[:clientReferenceInformation][:code] = options.dig(:order_id)
```